### ISSUE TITLE
pwm_dw driver does not compile

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#944]
**_Reported by Marcus Shawcroft:_**

pwm_dw driver uses the symbol PWM_DW_BASE_ADDR but there is no definition of that symbol in the zephyr source tree.

(Imported from Jira ZEP-1040)

**Comment by nashif** [comment]
**_by Marcus Shawcroft:_**

Root cause appears to be:

commit b0c0040894f224113a3ee98e5ea3bfc30161e538
Author: Anas Nashif <anas.nashif@intel.com>
Date:   Wed Apr 20 01:48:23 2016 -0400

    quark_se: quark_d2000: remove all defines that now come from qmsi
    
    QMSI defines all the registers of the SoC and other values in
    qm_soc_regs.h, so cleanup soc.h and rely on the data from the BSP
    contained in qm_soc_regs.h.
    
    Change-Id: I672925cf1c0144a9ed64073ea289c691285a8082
    Signed-off-by: Anas Nashif <anas.nashif@intel.com>

**Comment by nashif** [comment]
**_by Sharron LIU:_**

_Baohong Liu_ , _Anas Nashif_ , should dw driver adapt to this change?

**Comment by nashif** [comment]
**_by Baohong Liu:_**

_Sharron LIU_ _Anas Nashif_  My personal opinion is that there is no reason to use DW driver for quark.

**Comment by nashif** [comment]
**_by Marcus Shawcroft:_**

Is there any board supported by zephyr where the DW PWM driver works ?

**Comment by nashif** [comment]
**_by Anas Nashif:_**

the base address needs to be defined by a board that uses this driver, we do not have any board using it right now.

**Comment by nashif** [comment]
**_by Anas Nashif:_**

we will see of the EMSK can use this driver to support PWM

**Comment by nashif** [comment]
**_by Baohong Liu:_**

The driver is not being used by any platform right now. Anas is very right. When a new board needs this driver, the base address will be added into the board configuration. Since no action for now, let's close it.

**Comment by nashif** [comment]
**_by Baohong Liu:_**

Nothing to do for now.

**Comment by nashif** [comment]
**_by Sharron LIU:_**

_Marcus Shawcroft_ , could you review above comments from Anas and Baohong? You may reopen or report a new issue, if face any problems using DW driver @EMSK. Thanks.

**Comment by nashif** [comment]
**_by Marcus Shawcroft:_**

_Sharron LIU_  My interest in this driver arises from the various patches I've pushed up recently that impact the interface to all zephyr drivers.  In doing this work it is necessary to be able to compile / test all drivers.  It became apparent during this activity that a number of drivers are in the tree but for what ever reason, cannot be tested / cannot be compiled.  This leads to a situation in which a structural change across the zephyr tree cannot be adequately tested /  parts of the zephyr tree start to bit rot.  This issue is not specific to pwm-dw. pdwm-dw just happened to be the first instance of a driver where I found this issue, hence I raised the ticket.

I've posted a more general question, orthogonal to pwm-dw about how we handle unsused code in the zephyr tree on the list here:

https://lists.zephyrproject.org/archives/list/devel@lists.zephyrproject.org/thread/VSAOM5KXQAK5MGYZDK44EYYP5MNPN4UL/
